### DISCUSSION TITLE
CoilImage 2.3.5 Reopen the application and download the image again

### DISCUSSION THREAD

**Discussion by coder7iang** [skydoves/landscapist#553] [General]
指条明路？

**Comment by skydoves** [comment]
Hi @coder7iang, can you expand on this issue? Is that issue also happening on the [demo project](https://github.com/skydoves/landscapist/tree/main/app) on your device?

**Reply by coder7iang** [reply]
没有缓存，重新启动app会重新下载图片

CoilImage(
                imageModel = {
                    geekUserInfo?.baseInfo?.avatar
                },
                previewPlaceholder = painterResource(id = R.mipmap.base_avatar_placeholder ),
                component = rememberImageComponent {
                    +PlaceholderPlugin.Loading(painterResource(id = R.mipmap.base_avatar_placeholder))
                    +PlaceholderPlugin.Failure(painterResource(id = R.mipmap.base_avatar_placeholder))
                },
                imageOptions = ImageOptions(
                    contentScale = ContentScale.Crop,
                    contentDescription = "Avatar"
                ),
                modifier = Modifier
                    .size(70.dp)
                    .clip(RoundedCornerShape(35.dp)),
            )

**Reply by skydoves** [reply]
Check out the [Coil's documentation](https://coil-kt.github.io/coil/image_loaders/#caching). You should apply the cache strategy, and load the image with `ImageLoader` while loading the `CoilImage` as a parameter.

**Reply by coder7iang** [reply]
memoryCache 我试过没用  我已经替换GlideImage  没有这个问题了

**Reply by skydoves** [reply]
Try this:

```kotlin
CoilImage(
  imageRequest = {
      ImageRequest.Builder(LocalContext.current)
        .data(poster.poster)
        .crossfade(true)
        .build() },
  imageLoader = {
      ImageLoader.Builder(LocalContext.current)
           .memoryCache {
        MemoryCache.Builder(context)
            .maxSizePercent(0.25)
            .build()
    }
    .diskCache {
        DiskCache.Builder()
            .directory(context.cacheDir.resolve("image_cache"))
            .maxSizePercent(0.02)
            .build()
    }
        .build() },
  modifier = modifier,
)
```

**Reply by coder7iang** [reply]
I'll give it a try, thank you
